cellpack/autopack/DBRecipeHandler.py

```diff
@@ -376,6 +378,37 @@ def should_write(self, db, grad_name):
         return None, None
 
 
+class ResultDoc:
+    def __init__(self, db):
+        self.db = db
+
+    def handle_expired_results(self):
+        """
+        Check if the results in the database are expired and delete them if the linked object expired.
+        """
+        current_utc = datetime.now(timezone.utc)
+        results = self.db.get_all_docs("results")
+        if results:
+            for result in results:
+                result_data = self.db.doc_to_dict(result)
+                result_age = current_utc - result_data["timestamp"]
+                if result_age.days > 180 and not self.validate_existence(
+                    result_data["url"]
+                ):
+                    self.db.delete_doc("results", self.db.doc_id(result))
+            print("Results cleanup complete.")
+        else:
+            print("No results found in the database.")
```

Comment:
Do you have a preference between `print` and `logging` for this repo? I saw in `FirebaseHandler`, you're changing some `print`s to `logging.info`s, so I was just curious

Reply:
Good question! I'd say we should use `logging` over `print` for most scenarios, but for statements like these, I lean towards a cleaner message like `Results cleanup complete.` rather than the logging `2024-03-26 10:57:37,029 - root - INFO - FirebaseHandler.py:205 -           delete_doc(): successfully deleted path`.

I'm open to suggestions and happy to hear better approaches on this!

Reply:
Gotcha, that makes sense to me!